site-src/geps/gep-724.md

```diff
@@ -0,0 +1,624 @@
+bound to all Routes in the same namespace. Although that made the getting
+started guide simple, it would inevitably lead to some unfortunate mistakes in
+the future. As we've already learned with Kubernetes, it's very difficult to
+recover from insecure defaults. Instead, it's much safer to start with more
+explicit configuration that demonstrates clear intent to connect resources.
+
+#### 4. We Need to Support non-Gateway Route Parents
+With the expansion of this API, it's clear that a Route may have non-Gateway
+parents. This may be other Routes, mesh implementations, or custom Gateways.
+Although none of these concepts are well specified at this point, making this
+change now will give us more flexibility in the future.
+
+#### 5. Users Want Control over the Gateways Their Routes Are Attached To
+Initial feedback we've received has shown that users want to have very clear
+control over the Gateways their Routes are attached to. Even in the case of
+Gateway replacement, many Route owners would prefer to be involved in the
+process.
+
+As we get more feedback and usage of the API, we may identify more users that
+are interested in the more advanced capabilities that some form of selection may
+enable, but at this point it's clear that a large portion of users value an
+explicit way to attach a Route to a Gateway.
+
+#### 6. We Need to Maintain a Handshake Between Gateways and Routes
+Of course we do still need a handshake that will enable cross-namespace
+references between Routes and Gateways. This proposal leaves in the core
+capabilities of the v1alpha1 API for this. Gateways can specify the namespaces
+they trust Routes to bind from, and Routes directly reference the Gateways they
+want to attach to. This is largely similar to the ReferencePolicy model proposed
+for Route->Service references, but is embedded within the Route and Gateway
+resources. The alternatives below explore what this could look like with
+ReferencePolicy.
+
+## API Changes
+
+The proposed changes here can be summarized as:
+
+* Remove Route selector from the RouteBindingSelector in Gateway listeners.
+* Replace Route kind and group with optional list of accepted kinds and groups
+  in RouteBindingSelector.
+* Rename RouteBindingSelector to ListenerRoutes.
+* Replace the 3 options from Route -> Gateway (All, FromList, SameNamespace)
+  with a reference list that supports arbitrary kinds.
+
+### Gateway
+
+On Gateway, the only change involves removing the Route selector field.
+Everything else remains the same.
+
+#### Removed
+
+```go
+  // Selector specifies a set of route labels used for selecting
+  // routes to associate with the Gateway. If this Selector is defined,
+  // only routes matching the Selector are associated with the Gateway.
+  // An empty Selector matches all routes.
+  //
+  // Support: Core
+  //
+  // +optional
+  Selector *metav1.LabelSelector `json:"selector,omitempty"`
```

Comment:
This should be indented (or the lines further down should be unindented).

Reply:
Updated, thanks!